src/components/settings.tsx

```diff
@@ -0,0 +1,98 @@
+import FormButton from "@/components/formButton";
+import { User } from "@/components/profileCard";
+import { useRef, useState, FormEvent } from "react";
+import SettingsForm from "./settingsForm";
+
+export default function Settings({ user }: { user: User }) {
+  const firstNameRef = useRef<HTMLInputElement>(null);
+  const lastNameRef = useRef<HTMLInputElement>(null);
+  const aboutRef = useRef<HTMLInputElement>(null);
+  const locationRef = useRef<HTMLInputElement>(null);
+
+  const [loading, setLoading] = useState(false);
+  const [success, setSuccess] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  
+  const firstNameConst = firstNameRef.current?.value;
+  const lastNameConst = lastNameRef.current?.value;
+  const aboutConst = aboutRef.current?.value;
+  const locationConst = locationRef.current?.value;
+  
+  const handleSubmit = async (e: FormEvent) => {
+    e.preventDefault();
+    const updatedData = {
+      firstName: firstNameConst,
```

Comment:
you don't need refs here, checkout the docs on how to read values of uncontrolled inputs https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/forms_and_events/
```
  onSubmit={(e: React.SyntheticEvent) => {
    e.preventDefault();
    const target = e.target as typeof e.target & {
      email: { value: string };
      password: { value: string };
    };
    const email = target.email.value; // typechecks!
    const password = target.password.value; // typechecks!
    // etc...
  }}
```